### ISSUE TITLE
[FEATURE REQUEST]: DateOnly and TimeOnly for the new net6 target

### ISSUE THREAD

**Issue by DanSmith** [G-Research/ParquetSharp#423]
### Is your feature request related to a problem? Please describe.

It would be convenient to use the built in dotnet DateOnly instead of the ParquetSharp.Date, as it would prevent conversions between the two when working across the library and other code.

### Describe the solution you'd like

The parquet logical DATE could map to DateOnly and the logical TIME could map to TimeOnly in the net6 or later target.

### Additional context

Hi @adamreeve, I see that you have added a net6 target last week. Would you be open to having support added for the DateOnly and TimeOnly that are now available in net6? If so would you prefer to add it or would you be willing to accept a PR?

**Comment by adamreeve** [comment]
Hi @DanSmith, thanks for the feature request, I think adding this makes a lot of sense. We're generally pretty open to accepting PRs but I'd like to add this myself. I don't think we should just use the new types with the net6 target as that could be a fairly disruptive breaking change, so I'd like to experiment with how best to add this in a backwards compatible way.

**Comment by DanSmith** [comment]
Sounds great, let me know if you need any feedback/testing I am happy to help.

Perhaps as part of looking at this backwards compatibility we could consider if TimeSpans could be stored as the 12 byte parquet INTERVAL logical type, differentiating them from TIME and TimeOnly.

**Comment by adamreeve** [comment]
I've opened PR #424 for this and would be happy to get some feedback thanks @DanSmith.

I don't think using the Interval logical type is a great idea. In theory it's a better semantic match for the TimeSpan type, but it doesn't support negative values and has quite an awkward representation. For these reasons the Arrow project doesn't use this logical type for their Duration type, which is probably closest to a TimeSpan, but instead just use int64 values and rely on the embedded Arrow schema to know that the value represents a duration.

We can get away with using the Time logical type for TimeSpans and handle negative values or values > 24 hours when only using ParquetSharp, but this does cause issues for some libraries when trying to read files written by ParquetSharp, as they might validate that values are non-negative and less than 24 hours.

I looked into this recently and added some extra documentation on workarounds for handling TimeSpan values outside of the 24 hour range (https://github.com/G-Research/ParquetSharp/blob/master/docs/TimeSpan.md), but didn't find a great solution to this problem.

All that said, I'd be open to accepting a PR that supported conversion to and from the Interval logical type, as I don't think we support reading interval data at all currently, but it probably shouldn't be used in the default type mapping when writing.
